sip-security/src/main/java/de/ikor/sip/foundation/security/authentication/SIPAuthProvidersExistCondition.java

```diff
@@ -0,0 +1,43 @@
+package de.ikor.sip.foundation.security.authentication;
+
+import static org.apache.commons.collections4.CollectionUtils.isEmpty;
+
+import de.ikor.sip.foundation.security.config.SecurityConfigProperties.AuthProviderSettings;
+import java.util.List;
+import org.springframework.boot.autoconfigure.condition.ConditionMessage;
+import org.springframework.boot.autoconfigure.condition.ConditionOutcome;
+import org.springframework.boot.autoconfigure.condition.SpringBootCondition;
+import org.springframework.boot.context.properties.bind.BindResult;
+import org.springframework.boot.context.properties.bind.Bindable;
+import org.springframework.boot.context.properties.bind.Binder;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+
+/** Condition class matching whether authentication providers are defined */
+public class SIPAuthProvidersExistCondition extends SpringBootCondition {
+  private static final Bindable<List<AuthProviderSettings>> PROVIDER_LIST =
+      Bindable.listOf(AuthProviderSettings.class);
+
+  @Override
+  public ConditionOutcome getMatchOutcome(
+      ConditionContext context, AnnotatedTypeMetadata metadata) {
+    ConditionOutcome outcome = ConditionOutcome.match();
+    String listPropertyName = "sip.security.authentication.auth-providers";
+
+    if (isEmpty(getAuthProviderSettingsList(context, listPropertyName))) {
+      outcome =
+          ConditionOutcome.noMatch(
+              ConditionMessage.forCondition("SIP auth providers condition")
+                  .didNotFind("property")
+                  .items(listPropertyName));
+    }
+    return outcome;
+  }
+
+  private List<AuthProviderSettings> getAuthProviderSettingsList(
+      ConditionContext context, String listPropertyName) {
```

Comment:
Sorry for a bit latter idea :)
This peace of code is repeated in SIPAuthProviderCondition. 
Given that it concerns AuthProviderSettings, could we encapsulate it as a static method od AuthProviderSettings class? 
The return type would be Collection<AuthProviderSettings>. It would take Environment as an argument and  listPropertyName should be private field of AuthProviderSettings class, and SIPAuthProviderCondition can be refactored as well.

Please check if this approach makes sense :)